vue/src/components/LocalLayers.vue

```diff
@@ -0,0 +1,131 @@
+<script setup lang="ts">
+import { computed, ref } from 'vue';
+import { addLocalMapLayer, removeLocalMapLayer } from '../actions/localMapLayers';
+import { state } from '../store';
+import { getGeoJSONBounds } from '../utils';
+import { FitBoundsEvent } from '../actions/map';
+
+const isLoading = ref(false);
+
+async function loadFiles(files: File[]) {
+  const promise = Promise.all(files.map(async (file) => {
+    return new Promise<string>((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onload = function() {
+        resolve(reader.result as string);
+      };
+      reader.onerror = reject;
+      reader.readAsText(file, 'utf-8');
+    });
+  }));
+
+  isLoading.value = true;
+  promise.finally(() => {
+    isLoading.value = false;
+  });
+
+  const geoJsons = await promise;
+
+  geoJsons.forEach((data) => {
+    addLocalMapLayer(JSON.parse(data));
+  });
+}
+
+async function promptForGeoJSONs() {
+  const files = await new Promise<File[]>((resolve) => {
+    const fileEl = document.createElement('input');
+    fileEl.setAttribute('type', 'file');
+    fileEl.setAttribute('multiple', 'multiple');
+    fileEl.setAttribute('accept', '.geojson');
+    fileEl.addEventListener('change', () => {
+      const files = Array.from(fileEl.files ?? []);
+      resolve(files);
+    });
+    fileEl.click();
+  });
+
+  await loadFiles(files);
+}
+
+function deleteLayer(layerId: number) {
+  removeLocalMapLayer(layerId);
+}
+
+function focusLayer(layerId: number) {
+  const layer = state.localMapFeatureById[layerId];
+  if (!layer) return;
+
+  const bounds = getGeoJSONBounds(layer.geojson);
+  FitBoundsEvent.trigger(bounds);
+}
+
+const localLayers = computed(() => {
+  return state.localMapFeatureIds.map(
+    (id) => state.localMapFeatureById[id],
+  );
+})
+
+</script>
+
+<template>
+  <div class="h-100 overflow-y-auto">
+    <div class="text-center">
+      <v-btn
+        variant="flat"
+        color="secondary"
+        :disabled="isLoading"
+        :loading="isLoading"
+        @click="promptForGeoJSONs"
+      >
+        <template #prepend>
+          <v-icon>mdi-plus</v-icon>
+        </template>
+        Add local .geojson file
+      </v-btn>
+      <div
+        v-if="!localLayers.length"
+        class="mt-8 text-grey-darken-2"
+      >
+        No local layers
+      </div>
+    </div>
+    <v-list>
+      <v-list-item
+        v-for="layer in localLayers"
+        :key="layer.id"
+        :title="`Layer ${layer.id}`"
```

Comment:
I think it's fine for a Layer to begin with a Static Id like this when initialized, it may be nice to have an edit field that way a user can upload multiple layers and give them some reasonable name.  Just an edit button and allowing the user to rename locally for organizational purposes.

Reply:
I opted to just use an editable box in place of the static name.